agent/structs/structs.go

```diff
@@ -1466,6 +1465,11 @@ func (s *NodeService) Validate() error {
 		}
 	}
 
+	if s.Connect.Native && s.Port == 0 {
+		result = multierror.Append(result, fmt.Errorf(
+			"Connect native service must have non-zero port"))
+	}
+
 	return result
 }
```

Comment:
To note is this Validate() helper is used by both the [catalog](https://github.com/hashicorp/consul/blob/main/agent/consul/catalog_endpoint.go#L166) and [agent endpoints](https://github.com/hashicorp/consul/blob/main/agent/agent_endpoint.go#L1156).